### PR TITLE
Add golangci-lint linting using Github Actions and fix errors

### PR DESCRIPTION
**Pull request by twelho**
Note the version of golangci-lint is updated, and a couple of things move directory.

This is #214 rebased on top of master now that #172 is merged. It adds a [golangci-lint](https://github.com/golangci/golangci-lint) GitHub Action for running the linter on pushes/pull requests and fixes most of the linter errors (unused functions etc.) The rest are marked `TODO` for decisions at a later date.

### CODE REVIEW

File: pkg/utilities/version/version.go [hunk 1]
```diff
@@ -71,11 +71,9 @@ func LessThan(v1, v2 string) (bool, error) {
 }
 
 func parseVersion(v string) (int, int, int, error) {
```
Comment (luxas) [inline pkg/utilities/version/version.go:73]: This really should be using https://github.com/kubernetes/apimachinery/blob/master/pkg/util/version/version.go

File: cmd/wksctl/main.go [hunk 2]
```diff
@@ -45,8 +45,12 @@ func configureLogger(cmd *cobra.Command, args []string) {
 }
 
 func main() {
-	clusterv1.AddToScheme(scheme.Scheme)
-	baremetalv1.AddToScheme(scheme.Scheme)
+	if err := clusterv1.AddToScheme(scheme.Scheme); err != nil {
```
Comment (luxas) [inline cmd/wksctl/main.go:48]: I'd use `runtimeutil.Must()` here, but no need to change in this PR as I'll fix this up later

File: pkg/specs/validation_test.go [hunk 3]
```diff
@@ -120,6 +120,7 @@ spec:
   user: "vagrant"
 `
 
+//nolint:unused
```
Comment (luxas) [inline pkg/specs/validation_test.go:123]: Shall we open an issue to figure out how/if these should be used?

File: tools/go-lint [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 set -e
 
-if [ ! $(command -v golangci-lint) ]
+if [ ! -x "$(command -v golangci-lint)" ]
 then
     go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.17.1
```
Comment (luxas) [inline tools/go-lint:7]: This should be prob. upgraded to v1.27

File: pkg/utilities/version/version.go (hunk 1 above)
Comment (twelho) [reply pkg/utilities/version/version.go]: Changed, updated `LessThan` as well.

File: pkg/specs/validation_test.go (hunk 3 above)
Comment (twelho) [reply pkg/specs/validation_test.go:123]: I don't think it's important to track these in an issue right now, it's better to do that when we get to cleanup after the large structural changes have been merged.

**Pull request merged**